### PR TITLE
refactor: typescript implementation

### PR DESCRIPTION
**Pull request by melancholiai**

### CODE REVIEW

File: .nvmrc [hunk 1]
```diff
@@ -0,0 +1 @@
+v12
```
Comment (CptSchnitz) [inline .nvmrc]: v14

File: Dockerfile [hunk 2]
```diff
@@ -1,9 +1,11 @@
+ARG NODE_VERSION=14
+
 FROM ubuntu:20.04 AS builder
```
Comment (CptSchnitz) [inline Dockerfile]: better naming

File: package.json [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    "@aws-sdk/client-s3": "^3.39.0",
+    "@aws-sdk/lib-storage": "^3.48.0",
+    "@map-colonies/js-logger": "^0.0.3",
+    "axios": "^0.24.0",
+    "better-ajv-errors": "^1.1.2",
+    "concat-stream": "^2.0.0",
+    "config": "^3.3.6",
+    "http-status-codes": "^2.2.0",
+    "reflect-metadata": "^0.1.13",
+    "zx": "^4.2.0"
```
Comment (CptSchnitz) [inline package.json]: ?

File: src/s3client/s3Client.ts [hunk 4]
```diff
@@ -0,0 +1,126 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call */
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+/* eslint-disable @typescript-eslint/naming-convention */ // s3-client object commands arguments
```
Comment (CptSchnitz) [inline src/s3client/s3Client.ts]: ?

**Review by CptSchnitz — CHANGES_REQUESTED** [review]

File: .nvmrc (hunk 1 above)
Comment (melancholiai) [reply .nvmrc]: done.

File: Dockerfile (hunk 2 above)
Comment (melancholiai) [reply Dockerfile]: done.

File: package.json (hunk 3 above)
Comment (melancholiai) [reply package.json]: done.

File: src/s3client/s3Client.ts (hunk 4 above)
Comment (melancholiai) [reply src/s3client/s3Client.ts]: done.

**Pull request merged**